tests/manage/test_verify_all_fields_in_sc_yaml_with_oc_describe_sc.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import logging
+
+from tests import helpers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+from ocs_ci.framework.testlib import ManageTest, tier1
+from tests.fixtures import (
+    create_ceph_block_pool,
+    create_rbd_secret, create_cephfs_secret
+)
+
+log = logging.getLogger(__name__)
+
+SC_OBJ = None
+
+
+@tier1
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_cephfs_secret.__name__,
+    create_ceph_block_pool.__name__,
+)
+class TestVerifyAllFieldsInScYamlWithOcDescribe(ManageTest):
+    """
+    This class checks whether all the fields in the Storage Class
+    yaml matches oc describe sc output or not
+    """
+    @pytest.mark.parametrize(
+        argnames="interface",
+        argvalues=[
+            pytest.param(
+                *["RBD"], marks=pytest.mark.polarion_id("OCS-521")
+            ),
+            pytest.param(
+                *["CEPHFS"], marks=pytest.mark.polarion_id("OCS-522")
+            )
+        ]
+    )
+    def test_verify_all_fields_in_sc_yaml_with_oc_describe(self, interface):
+        """
+        Test function for RBD
+        """
+        if interface == "RBD":
+            log.info("Creating a RBD Storage Class")
+            self.sc_data = templating.load_yaml_to_dict(
+                constants.CSI_RBD_STORAGECLASS_YAML
+            )
+            self.sc_data['metadata']['name'] = (
+                helpers.create_unique_resource_name(
+                    'test', 'csi-rbd'
+                )
+            )
+        elif interface == "CEPHFS":
+            log.info("Creating a CephFS Storage Class")
+            self.sc_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_STORAGECLASS_YAML
+            )
+            self.sc_data['metadata']['name'] = (
+                helpers.create_unique_resource_name(
+                    'test', 'csi-cephfs'
+                )
+            )
```

Comment:
I think, that at least part of this code might be combined together. For
example the part with `helpers.create_unique_resource_name(...)` might be
easily replaced by general piece of code like this:

```python
        self.sc_data['metadata']['name'] = (
            helpers.create_unique_resource_name(
                'test', f'csi-{interface.lower()}'
            )
        )
```
And probably even whole `if...elif...` block of code might be replaced by
something like this:
```python
        log.info(f"Creating a {interface} Storage Class")
        self.sc_data = templating.load_yaml_to_dict(
            getattr(constants, f"CSI_{interface}_STORAGECLASS_YAML")
        )
        self.sc_data['metadata']['name'] = (
            helpers.create_unique_resource_name(
                'test', f'csi-{interface.lower()}'
            )
        )
```

Reply:
Really good comment. Thanks Fixed it.